### PR TITLE
BeyondJ Java Framework - The jar that will rule them all

### PR DESCRIPTION
**Pull request by nkasvosve**
BeyondJ is a new Java framework focused on ease of management, scalability and performance.

### CODE REVIEW

**Comment by blee-techempower** [comment]
Local test failed except query. Please resolve the issues.

File: deployment/vagrant-development/Vagrantfile [hunk 1]
```diff
@@ -9,6 +9,10 @@ check_provider_needs(provider)
 
 Vagrant.configure("2") do |config|
 
+  config.ssh.shell = "bash -c 'BASH_ENV=/etc/profile exec bash'"
+
+  #config.vm.synced_folder "/Users/qetscontractor/.m2/", "/home/vagrant/.m2/",create:false
+
```
Comment (msmith-techempower) [inline deployment/vagrant-development/Vagrantfile]: Can you remove these changes from your pull request?

If you have improvements to Vagrant, please make a separate PR so we can track them and the underlying reasons for them.

**Comment by msmith-techempower** [comment]
@nkasvosve Please add an entry in `.travis.yml` so that Travis-CI will automatically test your pull request.

**Comment by nkasvosve** [comment]
Done
Regards
 ________________________________
 


    On Thursday, December 10, 2015 5:02 PM, Mike Smith <notifications@github.com> wrote:
 

 @nkasvosve Please add an entry in .travis.yml so that Travis-CI will automatically test your pull request.—

**Comment by msmith-techempower** [comment]
@nkasvosve It seems that our latest changes in master are conflicting a bit here; would you mind rebasing and pushing to kick off another Travis build?

**Comment by nkasvosve** [comment]
I'm sorry I'm a bit lost. When I do a rebase GIt says there are no changes:
 git rebase origin/master
>>Current branch master is up to date.
Am I running the wrong command?

**Comment by msmith-techempower** [comment]
@nkasvosve I am no git expert, but [here is what StackOverflow suggests](http://stackoverflow.com/questions/7244321/how-to-update-a-github-forked-repository):

```bash
# Add the remote, call it "upstream":

git remote add upstream https://github.com/TechEmpower/FrameworkBenchmarks.git

# Fetch all the branches of that remote into remote-tracking branches,
# such as upstream/master:

git fetch upstream

# Make sure that you're on your master branch:

git checkout master

# Rewrite your master branch so that any commits of yours that
# aren't already in upstream/master are replayed on top of that
# other branch:

git rebase upstream/master
```

**Comment by nkasvosve** [comment]
Yes. Downloading by Travis takes a loooong time because of many dependencies. Trying alternative ways to speed up the startup. I'll figure out a suitable solution though

**Comment by msmith-techempower** [comment]
Good lord that's a lot of files; `beyondj-service.war` can be deleted, right? Same with `import.sql`?

**Comment by msmith-techempower** [comment]
Also, Travis failed a run:
```
/home/travis/build/TechEmpower/FrameworkBenchmarks/frameworks/Java/beyondj/setup.sh: line 14: cd: beyondj-launcher/deploy/: No such file or directory
```

File: frameworks/Java/beyondj/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+mvn clean install
+rm -rf beyondj-launcher/deploy/
+mkdir beyondj-launcher/deploy/
+
+cp beyondj-launcher/target/beyondj.jar beyondj-launcher/deploy/
+cd beyondj-launcher/deploy/
+jar xvf beyondj.jar
+rm beyondj.jar
+jar -cvf0M beyondj.jar META-INF *
+sudo cp beyondj.jar /var/www/html/beyondj.com/
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/build.sh]: I don't like this line - it should not require sudo and it should never install anything outside of `$IROOT`

File: frameworks/Java/beyondj/setup.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+fw_depends java8
+
+FILE=beyondj-launcher/deploy/beyondj.jar
+DIRECTORY=results
+
+if [ ! -d "$DIRECTORY" ]; then
+	mkdir results
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/setup.sh]: For what is this folder used?

File: frameworks/Java/beyondj/setup2.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+fw_depends java8 maven
+
+# rm beyondj-launcher/src/main/resources/launchers/webapp-launchers/beyondj-service.war
+sudo chmod -R 777 .
+mvn package
+# cp beyondj-service/target/beyondj-service.war beyondj-launcher/src/main/resources/launchers/webapp-launchers/
+# mvn clean package
+cp beyondj-launcher/target/beyondj.jar beyondj-launcher/deploy/
+
+java -jar beyondj-launcher/target/beyondj.jar system.platform.dbserver=${DBHOST} numInstances=10
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/setup2.sh]: This file is unused; please delete.

File: frameworks/Java/beyondj/old_setup.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	cd beyondj-launcher/deploy/
+	jar -cvf0M beyondj.jar META-INF *
+	cd ../../
+fi
+
+
+
+echo "Launching BeyondJ from location:$PWD"
+java -jar beyondj-launcher/deploy/beyondj.jar system.platform.dbserver=${DBHOST} numInstances=10
+
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/old_setup.sh]: This file is unused; please delete.

File: frameworks/Java/beyondj/build.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+cd ../../
+
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/build.sh]: This file also seems unused; can it be deleted?

File: frameworks/Java/beyondj/setup.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+fi
+
+cd beyondj-launcher/deploy/
+wget http://beyondj.com/beyondj.jar
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/setup.sh]: Please change `wget` to `fw_get`.

**Comment by msmith-techempower** [comment]
Other than the notes I provided, this is looking good and Travis-CI is agreeing with the build. Ready to merge as soon as those issues are cleared up.

File: frameworks/Java/beyondj/setup.sh [hunk 8]
```diff
@@ -0,0 +1,16 @@
+WORKING_DIR=beyondj-launcher/deploy
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/setup.sh]: The first line of this file should be `#!/bin/bash`

File: frameworks/Java/beyondj/setup.sh [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+#FILE=beyondj-launcher/deploy/beyondj.jar
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/setup.sh]: For the sake of keeping the filebase clean and small, please remove commented lines.

File: frameworks/Java/beyondj/setup.sh [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+if [ ! -d "$WORKING_DIR" ]; then
+        mkdir $WORKING_DIR
+fi
+
+#if [ ! -f "$FILE" ]; then
+        cd beyondj-launcher/deploy/
+        wget http://beyondj.com/beyondj.jar
```
Comment (msmith-techempower) [inline frameworks/Java/beyondj/setup.sh]: This line should be `fw_get -O http://beyondj.com/beyondj.jar`

**Comment by msmith-techempower** [comment]
Other than the notes above, all tests pass so we are ready to merge.

File: frameworks/Java/beyondj/setup.sh (hunk 10 above)
Comment (nkasvosve) [inline frameworks/Java/beyondj/setup.sh]: There is no problem with providing the SNAPSHOT version. I'll add version and update

**Comment by msmith-techempower** [comment]
Looks good; LGTM!
